lib/src/command/uploader.dart

```diff
@@ -28,8 +29,11 @@ class UploaderCommand extends PubCommand {
 
   UploaderCommand() {
     argParser.addOption('server',
-        defaultsTo: cache.sources.hosted.defaultUrl,
-        help: 'The package server on which the package is hosted.');
+        defaultsTo: Platform.environment['PUB_HOSTED_URL'] ??
+            'https://pub.dartlang.org',
+        help: 'The package server on which the package is hosted.\n'
+            'DEPRECATED: use `publish_to` in your pubspec.yaml or set the\n'
+            '\$PUB_HOSTED_URL environment variable.');
```

Comment:
No need to write that it's deprecated... just make it flag hidden. And then print as warning if `wasParsed('server') && isTerminal`...

Reply:
Done